File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> mp = new HashMap<>();
        for(String word:strs){
            char[] chars = word.toCharArray();
            Arrays.sort(chars);
            String sortedWord = new String(chars);
            if(!mp.containsKey(sortedWord))
                mp.put(sortedWord, new ArrayList<>());
            mp.get(sortedWord).add(word);
        }
        return new ArrayList<>(mp.values());
    }
}

File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
// groupAnagrams
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string, vector<string>> mp;
        for(int i=0; i<strs.size(); i++){
            string s = strs[i];
            sort(s.begin(), s.end());
            mp[s].push_back(strs[i]);
        }
        for(auto m:mp){
            ans.push_back(m.second);
        }
        return ans;
    }
};